### PR TITLE
[BUGFIX] Allow setting documents

### PR DESCRIPTION
**Pull request by georgringer**
To be in sync with old implementations it is helpful to e.g. remove the documents completly or just modify it how I like it

### CODE REVIEW

File: Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php [hunk 1]
```diff
@@ -74,6 +74,15 @@ public function getDocuments(): array
...
     }
+    
+    /**
+     * @param Document[] $documents
+     */
+    public function setDocuments(array $documents): void
+    {
+        $this->documents = $documents;
+    }
+
```
Comment (dkd-kaehm) [inline Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php:85]: setter on readonly prop?

**Review by dkd-kaehm — CHANGES_REQUESTED** [review]

File: Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php (hunk 1 above)
Comment (georgringer) [reply Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php:85]: it is $documents, not, $documents!

Comment (dkd-kaehm) [reply Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php:85]: > it is $documents, not, $documents!

What do you mean?

Comment (dkd-kaehm) [reply Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php:85]: > > it is $documents, not, $documents!
> 
> What do you mean?

Aaah OK, you mean docoment is read only not documents.

Comment (georgringer) [reply Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php:85]: correct! and I need to change the documents, e.g. setting it to an empty array if it shouldn't be indexed or whatsover

**Review by dkd-kaehm — APPROVED** [review]

**Pull request merged**